lib/Conch/Controller/User.pm

```diff
@@ -155,21 +173,48 @@ Delete a single setting for a user, provided it was set previously
 sub delete_setting ($c) {
 	my $key = $c->param('key');
 
-	my $user = Conch::Model::User->lookup( $c->stash('user_id') );
+	my $user = $c->stash('user');
 	Mojo::Exception->throw('Could not find previously stashed user')
 		unless $user;
 
-	my $settings = $user->settings;
+	my $count = $user->search_related('user_settings', { name => $key })->active->deactivate;
 
 	return $c->status( 404, { error => "No such setting '$key'" } )
-		unless $settings->{$key};
+		unless $count;
 
-	if ( $user->delete_setting($key) ) {
-		return $c->status(204);
-	}
-	else {
-		return $c->status( 500 => "Failed to delete setting" );
-	}
+	return $c->status(204);
+}
+
+=head2 change_password
+
+Stores a new password for the current user.
+
+Optionally takes a query parameter 'clear_tokens' (defaulting to true), to also revoke session
+tokens for the user, forcing all tools to log in again.
+
+=cut
+
+sub change_password ($c) {
+	my $body = $c->req->json;
+	return $c->status( 400, { error => 'Payload required' } ) unless $body;
```

Comment:
Let's move this to use the json validator like https://github.com/joyent/conch/blob/master/lib/Conch/Controller/DB/HardwareProduct.pm#L84

Reply:
```
    #          $got->{TEST} = '"TEST"'
    #     $expected->{TEST} = 'TEST'
```

Reply:
ok, I'm adding an entry to json-schema/input.yaml, and then this to `change_password`:

```
 sub change_password ($c) {
-       my $body = $c->req->json;
-       return $c->status( 400, { error => 'Payload required' } ) unless $body;
+       my $body =  $c->validate_input('UserPassword');
+       return if not $body;
 
        my $new_password = $body->{password};
-       return $c->status(400, { error => 'Missing parameter "password"' }) unless $new_password;
 
```